### PR TITLE
New `Cardano.Api.Query.Expr` module

### PR DESCRIPTION
**Pull request by newhoggy**
# Description

# Changelog

```yaml
- description: |
    Use query functions from the new export `Cardano.Api.Query.Expr` module instead of using
    the query data types directly. 
  compatibility: no-api-changes
  type: maintenance
```

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] The change log section in the PR description has been filled in
- [ ] New tests are added if needed and existing tests are updated.  These may include:
  - golden tests
  - property tests
  - roundtrip tests
  - integration tests
  See [Runnings tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [ ] Any changes are noted in the `CHANGELOG.md` for affected package
- [ ] The version bounds in `.cabal` files are updated
- [ ] CI passes. See note on CI.  The following CI checks are required:
  - [ ] Code is linted with `hlint`.  See `.github/workflows/check-hlint.yml` to get the `hlint` version
  - [ ] Code is formatted with `stylish-haskell`.  See `.github/workflows/stylish-haskell.yml` to get the `stylish-haskell` version
  - [ ] Code builds on Linux, MacOS and Windows for `ghc-8.10.7` and `ghc-9.2.7`
- [ ] Self-reviewed the diff

# Note on CI
If your PR is from a fork, the necessary CI jobs won't trigger automatically for security reasons.
You will need to get someone with write privileges.  Please contact IOG node developers to do this
for you.

### CODE REVIEW

File: configuration/cardano/mainnet-config.json [hunk 1]
```diff
@@ -1,6 +1,8 @@
 {
   "AlonzoGenesisFile": "mainnet-alonzo-genesis.json",
   "AlonzoGenesisHash": "7e94a15f55d1e82d10f09203fa1d40f8eede58fd8066542cf6566008068ed874",
+  "ApplicationName": "cardano-sl",
```
Comment (carbolymer) [inline configuration/cardano/mainnet-config.json]: Why it's needed?

File: configuration/cardano/mainnet-config.yaml [hunk 2]
```diff
@@ -30,6 +30,11 @@ LastKnownBlockVersion-Minor: 0
 LastKnownBlockVersion-Alt: 0
 MaxKnownMajorProtocolVersion: 2
 
+# In the Byron era some software versions are also published on the chain.
```
Comment (carbolymer) [inline configuration/cardano/mainnet-config.yaml]: Why it's needed?

**Review by carbolymer — APPROVED** [review]

File: configuration/cardano/mainnet-config.json (hunk 1 above)
Comment (erikd) [reply configuration/cardano/mainnet-config.json]: Did that not get removed?

File: configuration/cardano/mainnet-config.yaml (hunk 2 above)
Comment (newhoggy) [reply configuration/cardano/mainnet-config.yaml]: I think a rebase or cherry-pick picked up more than I intended.

File: configuration/cardano/mainnet-config.json (hunk 1 above)
Comment (newhoggy) [reply configuration/cardano/mainnet-config.json]: I think a rebase or cherry-pick picked up more than I intended.

Comment (newhoggy) [reply configuration/cardano/mainnet-config.json]: Fixed now.

File: configuration/cardano/mainnet-config.yaml (hunk 2 above)
Comment (newhoggy) [reply configuration/cardano/mainnet-config.yaml]: Fixed now.

**Pull request merged**